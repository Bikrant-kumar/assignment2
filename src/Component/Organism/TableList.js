import React from 'react';
import TableCard from '../Molecules/TableCard';
// import Title from '../Molecules/Title'
import Label from '../Atoms/Label';

function TableList(props) {
    return (
        <div>
            <Label text="TABLE"/>
            {/* <Title Name={"TABLE"}/> */}
            {
                props.tables && 
                props.tables.map((table) => <TableCard table={table} />)
            }
        </div>
    )
}

export default TableList

