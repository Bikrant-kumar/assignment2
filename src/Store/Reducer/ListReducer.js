export const initState = {    
   
};

export function itemReducer (state = initState, action) {
    switch (action.type) {
        case "DELETE_ITEM":
            const newItem = state.item.filter((item) => item.id !== action.id);
            return{
                ...state,
                item: newItem
            };

        case "DELETE_ROW":
            const newTable= action.table.item.splice(action.table.item.indexOf(action.rowId),1)
                action.table.totalCost = action.table.totalCost - action.rowCost;
                action.table.totalItem= action.table.totalItem- 1;
            return {
                ...state,
                ...newTable
            };
        
        case "ADD_ROW":
            const newRowInTable = action.table.item.push(action.menu)
            action.table.totalCost = action.table.totalCost + action.menu.Cost;
            action.table.totalItem= action.table.totalItem + 1;
            return {
                ...state,
                ...newRowInTable
            };

        case "ADD_ITEM":
            const items = [action.item, ...state.item]
            return {
                ...state,
                item: items,
            };
        case "ADD_REVIEW":
            const review = [action.review,...state.reviews]
            return{
                ...state,
                reviews:review
            };
        
        case "GET_STATE":
            return {  
                ...action.data,
            }

        case "DELETE_ALL_ITEM":
            const newTable1 = action.table.item.splice(0,action.table.item.length)
            action.table.totalCost = 0;
            action.table.totalItem = 0;
            return {
                ...state,
                ...newTable1
            }
            
        default:
            return state;    
    }
};

 export default itemReducer;