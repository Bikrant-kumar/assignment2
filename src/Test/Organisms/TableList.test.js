import React from 'react';
import { screen, render } from '@testing-library/react';
import TableList from '../../Component/Organism/TableList';
import ListReducer from '../../Store/Reducer/ListReducer';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MockData from '../MockData';
import { createStore } from 'redux';

describe('TableList',()=> {

    test('TableList Render',() => {   
        const tables= [
            {
              "id": "table-1",
              "Name": "Table1",
              "totalItem": 0,
              "totalCost": 0,
              "item": []
            },
            {
              "id": "table-2",
              "Name": "Table2",
              "totalItem": 0,
              "totalCost": 0,
              "item": []
            }] 
          render(
            <Provider store={createStore(ListReducer, MockData)}>
          <TableList tables={tables} />
          </Provider>);
          expect(screen.getByText("TABLE")).toBeInTheDocument();
       });
  
       test('test TableList with snapshot',()=>{
         const tables= [
                    {
                      "id": "table-1",
                      "Name": "Table1",
                      "totalItem": 0,
                      "totalCost": 0,
                      "item": []
                    },
                    {
                      "id": "table-2",
                      "Name": "Table2",
                      "totalItem": 0,
                      "totalCost": 0,
                      "item": []
                    }]
        const component= renderer.create(
          <Provider store={createStore(ListReducer, MockData)}><TableList tables={tables}/> </Provider>).toJSON();
        expect(component).toMatchSnapshot();
       });
  
  })