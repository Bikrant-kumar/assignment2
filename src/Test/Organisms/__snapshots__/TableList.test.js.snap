// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableList test TableList with snapshot 1`] = `
Array [
  <div>
    <div
      className="MuiBox-root MuiBox-root-6"
    >
      TABLE
    </div>
    <div
      className="MuiPaper-root MuiCard-root makeStyles-root-7 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        className="MuiButtonBase-root MuiCardActionArea-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="MuiCardContent-root"
        >
          <h2
            className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            onClick={[Function]}
          >
            <div
              className="MuiBox-root MuiBox-root-9"
            >
              Table1
            </div>
          </h2>
        </div>
        <span
          className="MuiCardActionArea-focusHighlight"
        />
      </button>
      <hr
        className="MuiDivider-root makeStyles-divider-8"
      />
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Total Item : 
        0
          | Total Cost : 
        0
      </p>
    </div>
    <div
      className="MuiPaper-root MuiCard-root makeStyles-root-7 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        className="MuiButtonBase-root MuiCardActionArea-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="MuiCardContent-root"
        >
          <h2
            className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            onClick={[Function]}
          >
            <div
              className="MuiBox-root MuiBox-root-10"
            >
              Table2
            </div>
          </h2>
        </div>
        <span
          className="MuiCardActionArea-focusHighlight"
        />
      </button>
      <hr
        className="MuiDivider-root makeStyles-divider-8"
      />
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Total Item : 
        0
          | Total Cost : 
        0
      </p>
    </div>
  </div>,
  " ",
]
`;
